subprojects/base-services/src/main/java/org/gradle/util/GUtil.java

```diff
@@ -156,20 +156,18 @@ public static boolean isTrue(@Nullable Object object) {
         return isTrue(object) ? object : defaultValue;
     }
 
-    public static <V, T extends Collection<? super V>> T addToCollection(T dest, boolean failOnNull, Iterable<? extends V>... srcs) {
-        for (Iterable<? extends V> src : srcs) {
-            for (V v : src) {
-                if (failOnNull && v == null) {
-                    throw new IllegalArgumentException("Illegal null value provided in this collection: " + src);
-                }
-                dest.add(v);
+    public static <V, T extends Collection<? super V>> T addToCollection(T dest, boolean failOnNull, Iterable<? extends V> src) {
+        for (V v : src) {
+            if (failOnNull && v == null) {
+                throw new IllegalArgumentException("Illegal null value provided in this collection: " + src);
             }
+            dest.add(v);
         }
         return dest;
     }
 
-    public static <V, T extends Collection<? super V>> T addToCollection(T dest, Iterable<? extends V>... srcs) {
-        return addToCollection(dest, false, srcs);
+    public static <V, T extends Collection<? super V>> T addToCollection(T dest, Iterable<? extends V> src) {
+        return addToCollection(dest, false, src);
     }
```

Comment:
Is this worth changing? I'm guessing some users are using this.

Reply:
Do you think so? It's not public API. I changed it because it was used nowhere and generic VarArgs are always problematic (at least if you follow the warnings). Calling such methods in a type safe way is not really possible and so calling them always leads to ugly code. Which does not feel right.

Alternatively, we can have two versions of the methods (we can add the one without varargs and leave the others). But that would leave dead code (from my perspective) lying around. Again, this is not public API.

Reply:
It's _old_ API, the package name doesn't include `internal` specifically, and the functionality is somewhat useful for a general use case. So yeah, I guess some people use it.

An alternative would be to deprecate and remove it.

That said, I'm fine removing it if you think it's best, just burnt myself with similar "obviously harmless" changes before and wanted to give a heads up. :)

Reply:
Looks like a general issue with this `org.gradle.util` package. We should address that. I'll follow up separately.

To detangle things: for this PR, I'll re-add the methods for now but mark them Deprecated.